Makefile

```diff
@@ -62,10 +62,11 @@ bindings-gen-go: build-contracts
 			continue; \
 		fi; \
 		type=$$(basename $$abi_file .abi.json); \
-		pkg=$$(basename $$abi_base .sol | tr "[:upper:]" "[:lower:]"); \
+		pkg=$$(basename $$type .sol | tr "[:upper:]" "[:lower:]"); \
 		mkdir -p ./bindings/go/$$pkg; \
 		abigen --abi $$abi_file --pkg $$pkg --type $$type --out ./bindings/go/$$pkg/$$type.go; \
 	done; \
+	go build ./... || exit 1; \
```

Comment:
nit: you need the `|| exit 1`. If `go build` fails, the make target will fail too.

Reply:
nit2: maybe add a message saying "running check sanity on bindings" or similar?

Reply:
(assuming you meant I _don't_ need the `exit 1` ) 

I think we actually do need this `exit 1` to cause the root workflow to fail ; not sure if this is default behavior or maybe it's an os thing but the make command doesn't fail of the go build fails without the exit 1 - I tried this myself through replacing the `go build ./...` with `go build ./.` to cause it to fail due to not finding any files to build. Running the failing command without the exit command still causes the overall make command to pass. 

Similar to the behavior we're seeing with abigen

Reply:
re second nit though I added messaging as you requested, thanks for the suggestion

Reply:
Sorry, I did mean "don't need" LOL

`Make` should propagate the errors thrown by shell commands executed by targets. Note that each line within a target is handled by a subshell. In your case you only have one command (single subshell) since you are concatenating commands like so `cmd0; cmd1; cmd2; ...` in which case you do have to exit

Here's a clear example 
```
[nicolas@macbook foo]# cat Makefile
fail:
        @go build ./.
shell:
        @echo hello; \
        go build ./.; \
        echo bye;
exit:
        @echo hello; \
        go build ./. || exit 1; \
        echo bye;

[nicolas@macbook foo]# make fail ; echo ">> $?"
package .: no Go files in /private/tmp/foo
make: *** [fail] Error 1
>> 2

[nicolas@macbook foo]# make shell ; echo ">> $?"
hello
package .: no Go files in /private/tmp/foo
bye
>> 0

[nicolas@macbook foo]# make exit ; echo ">> $?"
hello
package .: no Go files in /private/tmp/foo
make: *** [exit] Error 1
>> 2
```

Things like for loops need to be within a single subshell (so you need the `; \`) but other things could be in separate ones. Not a big deal though - it's ok as is.